doc/development.rst

```diff
@@ -195,18 +197,65 @@ Data management
 
 Data management should be isolated from data use as much as possible. This
 allows low-level physics classes to operate on references to data using the
-exact same device/host code.
-
-Container
-  Provide a CPU-based interface to allocate, store, and retrieve GPU data using
-  ``thrust`` vectors or other means. These can only be accessed via host code
-  but they use the CUDA API to manage data on the device. They can contain
-  metadata (string names, etc.) suitable for host-side debug output.
-
-View
-  A standalone plain-old-data reference to data owned by a container, used to
-  transfer GPU pointers and other kernel parameters between host and device.
-  The view's user should be closely linked with the Container: for example, the
-  device utility ``StackAllocator`` uses the ``StackAllocatorView`` to find
-  GPU-owned data managed by the ``StackAllocatorContainer``. Views can contain
-  other views (usually by value).
+exact same device/host code. Furthermore, state data (one per track) and
+shared data (definitions, persistent data, model data) should be separately
+allocated and managed.
+
+Store
+  Generic name for a class that manages GPU data by means of a host class,
+  using ``celeritas::DeviceVector`` (or ``thrust`` or ``VecGeom`` wrappers as
```

Comment:
Any preference or guidance on when to use celeritas classes to allocate device memory vs. thrust? I guess `celeritas::DeviceVector` is not really meant to be analogous to `thrust::device_vector` or `std::vector` (no dynamic resizing, etc.)?

Reply:
Good point, thanks. I've been using it for "setup-time" allocations where we only use plain old data and don't have to do resizing. I'll add that to the documentation.